### PR TITLE
Renamed 'Show Filters' button to 'Adjust Filters' and added sticky positioning 

### LINKED ISSUE #186: Make Show Filters Button always visible
When there are many offers, we need to scroll, and as a consequence, the `Show Filters` button is not visible. If the user wants to see the filters, he needs to scroll up, which can take a lot of time (for example, if he had scrolled 100 offers).

|   |   |
|---|---|
|  |  image |

### PR DESCRIPTION
**Pull request by pedrosilva17**
Closes #186.

### CODE REVIEW

File: src/components/HomePage/SearchResultsArea/SearchResultsWidget/OfferItemsContainer.js [hunk 1]
```diff
@@ -17,6 +17,8 @@ import clsx from "clsx";
 
 const useAdvancedSearchButtonStyles = makeStyles((theme) => ({
     root: {
+        top: "0",
+        position: "sticky",
```
Comment (TiagooGomess) [inline src/components/HomePage/SearchResultsArea/SearchResultsWidget/OfferItemsContainer.js:21]:
```suggestion
        top: 0,
        position: "sticky",
        zIndex: 1,
```

You can solve the problem discussed in https://github.com/NIAEFEUP/nijobs-fe/issues/186#issuecomment-1059385124 by adding the `zIndex: 1,` line, which makes the button be in the front of the offers.

However, make sure the offers behind the button are not visible.

Comment (pedrosilva17) [reply src/components/HomePage/SearchResultsArea/SearchResultsWidget/OfferItemsContainer.js:21]: Wow, it was that simple.
Regarding the last sentence, what you're suggesting is that I hide the offers dynamically as they go past the filters button?

Comment (pedrosilva17) [reply src/components/HomePage/SearchResultsArea/SearchResultsWidget/OfferItemsContainer.js:21]: This is the current behaviour of the button: It hides parts of the offer that should be behind it.
2022-03-04-21:26:13-screenshot

Comment (DoStini) [reply src/components/HomePage/SearchResultsArea/SearchResultsWidget/OfferItemsContainer.js:21]: If you don't want the button to stay on top of the offers, you don't need to hide them. You could simply put the button outside of the scroll area

Comment (imnotteixeira) [reply src/components/HomePage/SearchResultsArea/SearchResultsWidget/OfferItemsContainer.js:21]: The offers should not show behind the button, it should have a "clear area" around it. It looks broken this way

**Comment by imnotteixeira** [comment]
Try to be more specific when making PRs: which button? which name? which behavior?

PS: after looking at the commit/comments above I managed to find, but I (or anyone else) shouldn't have to look much in order to understand what a PR is about

**Comment by pedrosilva17** [comment]
Got it, will fix that tomorrow.

File: src/components/HomePage/SearchResultsArea/SearchResultsWidget/OfferItemsContainer.js (hunk 1 above)
Comment (pedrosilva17) [reply src/components/HomePage/SearchResultsArea/SearchResultsWidget/OfferItemsContainer.js:21]: After searching for hours trying to come up with a solution involving margins and padding (without success), I found out that adding `backgroundColor: "white"` colored the button header and now hides the offers behind it on scroll. Is this an acceptable solution?

```suggestion
        top: "0",
        position: "sticky",
        zIndex: 1,
        backgroundColor: "white",
```
Here's how it looks:

2022-03-13-14:29:04-screenshot

I also increased the padding at the bottom of the header to match the padding on top: It looks better with the header changes, in my opinion. What do you think?

Comment (TiagooGomess) [reply src/components/HomePage/SearchResultsArea/SearchResultsWidget/OfferItemsContainer.js:21]: At first sight, I would say it's acceptable, but please commit the changes to give better feedback.

Comment (pedrosilva17) [reply src/components/HomePage/SearchResultsArea/SearchResultsWidget/OfferItemsContainer.js:21]: Done. Sorry for taking so long, lack of experience made me overcomplicate (I think).

**Review by imnotteixeira — APPROVED** [review]

**Comment by imnotteixeira** [comment]
You'll need to rebase with the updated `develop` for the build to pass. If you don't know what any of this means, ask someone at the team

**Comment by pedrosilva17** [comment]
`! [rejected]        feature/adjust-filters-button-always-visible -> feature/adjust-filters-button-always-visible (stale info)` 
This happens when I do the last step (`git push origin <branch> --force-with-lease`. I tried adding `--prune` on the pull from develop and repeating the process, but the result is the same. What can I do?

Steps followed:
`git checkout develop`
`git pull origin develop`
`git checkout <branch>`
`git rebase develop`
`git push origin <branch> --force-with-lease`

**Comment by imnotteixeira** [comment]
> `! [rejected] feature/adjust-filters-button-always-visible -> feature/adjust-filters-button-always-visible (stale info)` This happens when I do the last step (`git push origin <branch> --force-with-lease`. I tried adding `--prune` on the pull from develop and repeating the process, but the result is the same. What can I do?
> 
> Steps followed: `git checkout develop` `git pull origin develop` (added `--prune` here on the second try) `git checkout <branch>` `git rebase develop` `git push origin <branch> --force-with-lease`

That indeed is weird. If the rebase succeeded, try running `git fetch` and retrying the `git push --force-with-lease`. Or else, you can restart the operation by running `git reset --hard origin/feature/adjust-filters-button-always-visible` **while on the `feature/adjust-filters-button-always-visible` branch** and then re-doing the rebase flow all over again. Maybe there was some problem with the github remote, and the upstream branch was gone for a while (?)

**Comment by pedrosilva17** [comment]
> > `! [rejected] feature/adjust-filters-button-always-visible -> feature/adjust-filters-button-always-visible (stale info)` This happens when I do the last step (`git push origin <branch> --force-with-lease`. I tried adding `--prune` on the pull from develop and repeating the process, but the result is the same. What can I do?
> > Steps followed: `git checkout develop` `git pull origin develop` (added `--prune` here on the second try) `git checkout <branch>` `git rebase develop` `git push origin <branch> --force-with-lease`
> 
> That indeed is weird. If the rebase succeeded, try running `git fetch` and retrying the `git push --force-with-lease`. Or else, you can restart the operation by running `git reset --hard origin/feature/adjust-filters-button-always-visible` **while on the `feature/adjust-filters-button-always-visible` branch** and then re-doing the rebase flow all over again. Maybe there was some problem with the github remote, and the upstream branch was gone for a while (?)

Using `git fetch` solved the problem. The tracking of the remote must have had issues somehow. Thanks!

**Comment by pedrosilva17** [comment]
Hm, spoke too soon.

**Comment by pedrosilva17** [comment]
Oh, I think I know what caused it, I forgot to change the test with the updated name of the button. My bad.

**Review by TiagooGomess — APPROVED** [review]

**Pull request merged**